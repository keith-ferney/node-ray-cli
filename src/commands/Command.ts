/* eslint-disable no-unused-vars */

import { classOf } from '@/lib/helpers';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export abstract class Command {
    public command = '';

    public help = '';

    public static instance: Command;

    public client: any = null;

    public uuid: string | null = null;

    public argv: Argv | null = null;

    constructor() {
        Command.instance = this;
    }

    public name(): string {
        return <string>this.command.split(' ').shift();
    }

    public builder = (): Record<string, any> => {
        return {};
    };

    public handle(argv: Argv) {
        this.argv = argv;
    }

    public displayUuid(instance: Ray) {
        if (this.argv === null) {
            return;
        }

        if (typeof this.argv['quiet'] === 'undefined' || this.argv['quiet'] === false) {
            console.log(instance.uuid);
        }
    }
}
