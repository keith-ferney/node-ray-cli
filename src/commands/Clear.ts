import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Clear extends Command {
    public override command = 'clear';
    public override help = 'clear the current Ray screen';

    public override handle(argv: Argv) {
        super.handle(argv);

        const instance = Ray.create(this.client, this.uuid);

        instance.clearScreen();
    }
}
