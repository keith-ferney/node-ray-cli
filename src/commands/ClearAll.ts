import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class ClearAll extends Command {
    public override command = 'clear-all';
    public override help = 'Clear current and all previous Ray screens';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid ?? argv['uuid'] ?? null);

        instance.clearAll();

        super.afterHandle(instance);
    }
}
