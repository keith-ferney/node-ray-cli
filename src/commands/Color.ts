import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Color extends Command {
    public override command = 'color <uuid> <color>';
    public override help = 'change to color of a previously sent payload';

    public override handle(argv: Argv) {
        super.handle(argv);

        const instance = Ray.create(this.client, argv['uuid']);

        instance.uuid = argv['uuid'];

        instance.color(argv['color']);

        this.displayUuid(instance);
    }
}
