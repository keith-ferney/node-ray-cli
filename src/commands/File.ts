import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class File extends Command {
    public override command = 'file <filename>';
    public override help = 'Display the contents of a file in Ray';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid ?? argv['uuid']);

        instance.file(argv['filename']);

        super.displayUuid(instance);

        super.afterHandle(instance);
    }
}
