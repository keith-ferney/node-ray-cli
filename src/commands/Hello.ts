import { Command } from './Command';
import { Argv } from 'yargs';

export class Hello extends Command {
    public override command = 'hello <name>';
    public override help = 'say hello to someone';

    public override handle(argv: Argv) {
        console.log(`hello ${argv.name}!`);
    }
}
