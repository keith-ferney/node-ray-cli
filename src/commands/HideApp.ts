import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class HideApp extends Command {
    public override command = 'hide-app';
    public override help = 'Hide the Ray application';

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid);

        instance.hideApp();
    }
}
