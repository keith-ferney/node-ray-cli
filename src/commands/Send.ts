import { Command } from './Command';
import { Argv } from 'yargs';
import { Ray } from 'node-ray';

export class Send extends Command {
    public override command = 'send <data>';
    public override help = 'send some data to Ray';

    public builder = () => {
        return {
            red: { default: false },
            blue: { default: false },
            green: { default: false },
            orange: { default: false },
            purple: { default: false },
            gray: { default: false },
            hide: { default: false },
        };
    };

    public override handle(argv: Argv) {
        if (super.handle(argv) === false) {
            return false;
        }

        const instance = Ray.create(this.client, this.uuid).send(argv['data']);

        if (typeof argv['hide'] !== 'undefined' && <boolean>(<unknown>argv['hide']) === true) {
            instance.hide();
        }

        const colorNames = ['red', 'blue', 'green', 'orange', 'purple', 'gray'];

        colorNames.forEach(colorName => {
            if (typeof argv[colorName] !== 'undefined' && argv[colorName] === true) {
                instance.color(colorName);
            }
        });

        super.displayUuid(instance);
    }
}
