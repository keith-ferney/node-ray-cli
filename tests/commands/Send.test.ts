/* eslint-disable no-undef */
import { Send } from '@/commands/Send';
import { FakeClient } from '@tests/TestClasses/FakeClient';
import { Argv } from 'yargs';

let client: FakeClient;
let command: Send;

beforeEach(() => {
    client = new FakeClient();

    command = new Send();
    command.uuid = 'fakeUuid';
    command.client = client;
});

it('sends a payload', () => {
    command.handle(<Argv>(<unknown>{ _: ['send'], quiet: true, data: 'test data' }));

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a payload with color', () => {
    command.handle(<Argv>(<unknown>{ _: ['send'], quiet: true, data: 'test data', red: true }));

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a collapsed payload', () => {
    command.handle(<Argv>(<unknown>{ _: ['send'], quiet: true, data: 'test data', hide: true }));

    expect(client.sentPayloads()).toMatchSnapshot();
});
