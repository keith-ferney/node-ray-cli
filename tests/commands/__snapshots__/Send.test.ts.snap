// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends a collapsed payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "test data",
          ],
        },
        "origin": Object {
          "file": "/src/commands/Send.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {},
        "origin": Object {
          "file": "/src/commands/Send.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "hide",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "test data",
          ],
        },
        "origin": Object {
          "file": "/src/commands/Send.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends a payload with color 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "values": Array [
            "test data",
          ],
        },
        "origin": Object {
          "file": "/src/commands/Send.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "log",
      },
    ],
    "uuid": "fakeUuid",
  },
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "color": "red",
        },
        "origin": Object {
          "file": "/tests/Ray.test.ts",
          "function_name": "xxxx",
          "hostname": "fake-host",
          "line_number": 999,
        },
        "type": "color",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;
